lib/sinon/spy.js

```diff
@@ -53,9 +53,11 @@ function matchingFake(fakes, args, strict) {
         return undefined;
     }
 
-    for (var i = 0, l = fakes.length; i < l; i++) {
-        if (fakes[i].matches(args, strict)) {
-            return fakes[i];
+    var fakesIndex = fakes.length;
+
+    while (fakesIndex--) {
```

Comment:
Yeah, the filter is probably more intuitive, I wasn't aware which target we we're aiming for so I stuck with a known approach since I knew it supposed to run in a browser.

The last `|| undefined` seems unecessary though since pop() returns undefined by default. 

Do you want me to make those changes and update the PR?